File: Dsa/weekly-practice/practiceday9.py
def isAnagram(s1, s2):
    freq={}
    if len(s1) != len(s2):
        return False
    for i in s1:
        if i in freq:
            freq[i]+=1
        else:
            freq[i]=1
    for i in s2:
        if i not in freq or freq[i]==0:
            return False
        freq[i]-=1
    return True
print(isAnagram("appple", "ppplea"))

def removeDuplicate(char):
    freq={}
    total=""
    for i in char:
        if i not in freq:
            freq[i]=True
            total+=i
        
    return total
print(removeDuplicate("apple"))

File: Dsa/weekly-practice/practiceday9.js
function isAngram(s1, s2){
    if(s1.length !== s2.length){
        return false
    }
    let freq={}

    for(let i of s1){
        freq[i]=(freq[i]|0)+1
    }
    for (let i of s2){
        if(!freq[i]){
            return false
        }
        freq[i]--
    }
    return true
}
console.log(isAngram("apple", "palea"))

function removeDuplicate(str){
    let freq={}
    let result=""
    for(let i of str){
        if(!freq[i]){
            freq[i]=true
            result+=i
        }
    }
    return result
}
console.log(removeDuplicate("apple"))